Tests/ExtractionTests/ExtractionTests.swift

```diff
@@ -1,168 +1,195 @@
+                let id = newProposal.id
+                guard !id.isEmpty  else {
+                    continue
+                }
+
+                guard let sourceProposal = expectedResultsByProposalID[id] else {
+                    Issue.record("Unable to find id \(id) in expected results.")
+                    continue
+                }
+
+                #expect(newProposal == sourceProposal)
             }
-            
-            XCTAssertEqual(newProposal, sourceProposal)
         }
-        
+
         // Check the generated JSON to catch issues such as removing properties from the schema
-        let expectedResultsURL = snapshotURL.appending(path: "expected-results.json")
-        let expectedJSONData = try Data(contentsOf: expectedResultsURL)
-        let actualJSONData = try extractedEvolutionMetadata.jsonRepresentation
-        XCTAssertEqual(expectedJSONData, actualJSONData)
-        
-        // Uncomment to write full JSON files out to compare in a diff tool
-        // try writeJSONFilesToPath(expected: expectedJSONData, actual: actualJSONData, path: "~/Desktop")
+        @Test func `Serialization`() throws {
+            let expectedResultsURL = snapshotURL.appending(path: "expected-results.json")
+            let expectedJSONData = try Data(contentsOf: expectedResultsURL)
+            let actualJSONData = try extractedEvolutionMetadata.jsonRepresentation
+            #expect(expectedJSONData == actualJSONData)
 
+            // Uncomment to write full JSON files out to compare in a diff tool
+            // try writeJSONFilesToPath(expected: expectedJSONData, actual: actualJSONData, path: "~/Desktop")
+        }
     }
-    
-    func testWarningsAndErrors() async throws {
-        
-        let snapshotURL = try urlForSnapshot(named: "Malformed")
-        let extractionJob = try await ExtractionJob.makeExtractionJob(from: .snapshot(snapshotURL), output: .none, ignorePreviousResults: true)
-        let expectedResults = try XCTUnwrap(extractionJob.expectedResults, "No expected results found in bundle '\(snapshotURL.absoluteString)'")
-        
-        // VALIDATION ENHANCEMENT: Tools like Xcode like to add a newline character to empty files
-        // Possibly instead of using 0007-empty-file.md to test, test separately?
-        // Or test that the file hasn't been corrupted. (Can you even check it into github?)
 
-        let extractionMetadata = try await EvolutionMetadataExtractor.extractEvolutionMetadata(for: extractionJob)
-        
-        // This test zips the extraction results with the expected results
-        // If the two arrays don't have the same count, the test data itself has an error
-        XCTAssertEqual(extractionMetadata.proposals.count, expectedResults.proposals.count)
-        
-        for (actualResult, expectedResult) in zip(extractionMetadata.proposals, expectedResults.proposals) {
-            XCTAssertEqual(actualResult, expectedResult)
+    struct `Malformed proposals` {
```

Comment:
If the array count check is folded into `Warnings and errors()` then this may not need to be a separate suite.

```suggestion
   @Suite
   struct `Malformed Proposals` {
```